server/internal/core/application/indexer.go

```diff
@@ -0,0 +1,553 @@
+}
+
+type indexerService struct {
+	pubkey *secp256k1.PublicKey
+
+	repoManager ports.RepoManager
+}
+
+func NewIndexerService(
+	pubkey *secp256k1.PublicKey,
+	repoManager ports.RepoManager,
+) IndexerService {
+	return &indexerService{
+		pubkey:      pubkey,
+		repoManager: repoManager,
+	}
+}
+
+func (i *indexerService) GetCommitmentTxInfo(
+	ctx context.Context, txid string,
+) (*CommitmentTxResp, error) {
+	round, err := i.repoManager.Rounds().GetRoundWithTxid(ctx, txid)
+	if err != nil {
+		return nil, err
+	}
+
+	leaves := round.VtxoTree.Leaves()
+
+	vtxo, err := i.repoManager.Vtxos().GetVtxos(
+		ctx,
+		[]domain.VtxoKey{
+			{
+				Txid: leaves[0].Txid,
+				VOut: 0,
+			},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		totalBatchAmount   uint64 = 0
+		totalForfeitAmount uint64 = 0
+		totalInputVtxos    int32  = 0
+	)
+	for _, request := range round.TxRequests {
+		for _, input := range request.Inputs {
+			totalForfeitAmount += input.Amount
+		}
+		totalInputVtxos += int32(len(request.Inputs))
+
+		for _, receiver := range request.Receivers {
+			if !receiver.IsOnchain() {
+				totalBatchAmount += receiver.Amount
+			}
+		}
+	}
+
+	batches := make(map[VOut]Batch)
+	// TODO: currently commitment tx has only one batch, in future multiple batches will be supported
+	batches[0] = Batch{
+		TotalBatchAmount:   totalBatchAmount,
+		TotalForfeitAmount: totalForfeitAmount,
+		TotalInputVtxos:    totalInputVtxos,
+		TotalOutputVtxos:   int32(len(leaves)),
+		ExpiresAt:          vtxo[0].ExpireAt,
+		Swept:              round.Swept,
+	}
+
+	return &CommitmentTxResp{
+		StartedAt: round.StartingTimestamp,
+		EndAt:     round.EndingTimestamp,
+		Batches:   batches,
+	}, nil
+}
+
+func (i *indexerService) GetVtxoTree(ctx context.Context, req VtxoTreeReq) (*VtxoTreeResp, error) {
+	vtxoTree, err := i.repoManager.Rounds().GetVtxoTreeWithTxid(ctx, req.BatchOutpoint.Txid) //TODO repo methods needs to be updated with multiple batches in future
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, pageResp := paginate(flattenNodes(vtxoTree), req.Page, maxPageSizeVtxoTree)
+
+	return &VtxoTreeResp{
+		Nodes: nodes,
+		Page:  pageResp,
+	}, nil
+}
+
+func (i *indexerService) GetForfeitTxs(ctx context.Context, req ForfeitTxsReq) (*ForfeitTxsResp, error) {
+	round, err := i.repoManager.Rounds().GetRoundWithTxid(ctx, req.BatchOutpoint.Txid) //TODO batch thing
+	if err != nil {
+		return nil, err
+	}
+
+	txs := make([]string, 0, len(round.ForfeitTxs))
+	for _, tx := range round.ForfeitTxs {
+		txs = append(txs, tx.Txid)
+	}
+
+	forfeitTxs, pageResp := paginate(txs, req.Page, maxPageSizeForfeitTxs)
+
+	return &ForfeitTxsResp{
+		Txs:  forfeitTxs,
+		Page: pageResp,
+	}, nil
+
+}
+
+func (i *indexerService) GetConnectors(ctx context.Context, req ConnectorsReq) (*ConnectorResp, error) {
+	round, err := i.repoManager.Rounds().GetRoundWithTxid(ctx, req.BatchOutpoint.Txid) //TODO batch thing
```

Comment:
same here, we can make use of an ad hoc `GetRoundConnectorTree`

Reply:
Done